src/main/java/LinkedQueue.java

```diff
@@ -0,0 +1,125 @@
+import java.util.NoSuchElementException;
+
+public class LinkedQueue<T> implements Queue<T> {
+
+    private Node<T> front;
+    private Node<T> rear;
+    private int size;
+
+    public LinkedQueue() {
+        front = null;
+        rear = null;
+        size = 0;
+    }
+
+    @Override
+    public void enqueue(T element) {
+        Node<T> toEnqueue = new Node<>(element);
+
+        // If nothing is in the queue, then the front is null
+        // and the enqueued element will become the front,
+        // otherwise, the enqueued element is added after the
+        // current rear.
+        if (isEmpty()) {
+            front = toEnqueue;
+        }
+        else {
+           rear.setNext(toEnqueue);
+        }
+        // Either way, the rear of the queue will be the newly
+        // enqueued element.
+        rear = toEnqueue;
+        size++;
+    }
+
+    @Override
+    public T dequeue() {
+        if (isEmpty()) {
+            throw new NoSuchElementException("Dequeueing from an empty queue.");
+        }
```

Comment:
The exception message here is kind of useless. "No such element" already tells you the queue was empty, and the stack trace will tell you that you were in `dequeue`. It doesn't hurt, but it doesn't help.

Reply:
Message is just to show them that it's a thing.